libs/services/designer-client-services/src/lib/standard/run.ts

```diff
@@ -0,0 +1,188 @@
+import { inputsResponse, outputsResponse } from '../__test__/__mocks__/monitoringInputsOutputsResponse';
+import type { HttpRequestOptions, IHttpClient } from '../httpClient';
+import type { IRunService } from '../run';
+import type { CallbackInfo } from '../workflow';
+import type { Runs, Run, RunError, ContentLink, BoundParameters } from '@microsoft/designer-ui';
+import { isCallbackInfoWithRelativePath, getCallbackUrl } from '@microsoft/designer-ui';
+import type { ArmResources } from '@microsoft/utils-logic-apps';
+import { ArgumentException, HTTP_METHODS, UnsupportedException } from '@microsoft/utils-logic-apps';
+
+export interface RunServiceOptions {
+  apiVersion: string;
+  baseUrl: string;
+  httpClient: IHttpClient;
+  accessToken?: string;
+  workflowName: string;
+  isDev?: boolean;
+}
+
+export class StandardRunService implements IRunService {
+  _isDev = false;
+
+  constructor(public readonly options: RunServiceOptions) {
+    const { apiVersion, baseUrl, isDev } = options;
+    if (!baseUrl) {
+      throw new ArgumentException('baseUrl required');
+    } else if (!apiVersion) {
+      throw new ArgumentException('apiVersion required');
+    }
+    this._isDev = isDev || false;
+  }
+
+  async getContent(contentLink: ContentLink): Promise<any> {
+    const { uri } = contentLink;
+    const { httpClient } = this.options;
+
+    if (!uri) {
+      throw new Error();
+    }
+    const response = await httpClient.get<any>({
+      uri,
+    });
+    return response;
+
+    /*const contentType = response.headers.get('Content-Type');
+    if (contentType?.startsWith('application/json')) {
+      return response.json();
+    } else if (contentType?.startsWith('text/')) {
+      return response.text();
+    } else if (contentType?.startsWith('application/octet-stream')) {
+      return response.blob();
+    } else if (contentType?.startsWith('multipart/form-data')) {
+      return response.formData();
+    } else {
+      return response.arrayBuffer();
+    }*/
+  }
+
+  private getAccessTokenHeaders = () => {
+    const { accessToken } = this.options;
+    if (!accessToken) {
+      return undefined;
+    }
+
+    return new Headers({
+      Authorization: accessToken,
+    });
+  };
+
+  async getMoreRuns(continuationToken: string): Promise<Runs> {
```

Comment:
Our search service has a couple methods set up already to handle requests with continuation tokens too, this may not be for this PR but we should at some point pull those out to a generic file so they can use the same logic (they're essentially doing the same thing just with different data types).  If you want to look at them they are `getAzureResourceRecursive` for requesting in series and `batchAzureResourceRequests` for requesting in parallel

Reply:
Thanks! Yeah I was planning on doing a separate PR for the run service.